### PR TITLE
Allow populating bearer token from file contents

### PR DESCRIPTION
**Pull request by liggitt**

### CODE REVIEW

**Comment by liggitt** [comment]
Related to #2530, adds token population capability first, hopefully so it makes it into a tagged release prior to #2530 merging

**Comment by liggitt** [comment]
@deads2k split out token population to get that available in the docker deployer image early

File: pkg/cmd/util/clientcmd/clientcmd.go [hunk 1]
```diff
@@ -68,6 +71,13 @@ func (cfg *Config) bindEnv() {
 	if value, ok := util.GetEnv("BEARER_TOKEN"); ok && len(cfg.CommonConfig.BearerToken) == 0 {
 		cfg.CommonConfig.BearerToken = value
 	}
+	if value, ok := util.GetEnv("BEARER_TOKEN_FILE"); ok && len(cfg.CommonConfig.BearerToken) == 0 {
+		if tokenData, err := ioutil.ReadFile(value); err == nil {
+			cfg.CommonConfig.BearerToken = strings.TrimSpace(string(tokenData))
+		} else {
+			glog.Error(err)
```
Comment (deads2k) [inline pkg/cmd/util/clientcmd/clientcmd.go]: Why shouldn't this method return an error and leave the handling up the caller?  It looks like a utility and the intent of the method failed.

File: pkg/cmd/util/clientcmd/clientcmd.go [hunk 2]
```diff
@@ -68,6 +71,13 @@ func (cfg *Config) bindEnv() {
 	if value, ok := util.GetEnv("BEARER_TOKEN"); ok && len(cfg.CommonConfig.BearerToken) == 0 {
 		cfg.CommonConfig.BearerToken = value
 	}
+	if value, ok := util.GetEnv("BEARER_TOKEN_FILE"); ok && len(cfg.CommonConfig.BearerToken) == 0 {
+		if tokenData, err := ioutil.ReadFile(value); err == nil {
+			cfg.CommonConfig.BearerToken = strings.TrimSpace(string(tokenData))
```
Comment (deads2k) [inline pkg/cmd/util/clientcmd/clientcmd.go]: Seems backwards to allow the file to override the direct setting of the token.  I could see either having the priority the other way around or having an error if both are specified.

Comment (liggitt) [inline pkg/cmd/util/clientcmd/clientcmd.go]: it doesn't... it only processes the file if there's no token already in the config

File: pkg/cmd/util/clientcmd/clientcmd.go (hunk 1 above)
Comment (liggitt) [inline pkg/cmd/util/clientcmd/clientcmd.go]: I guess I'm ok with returning an error, though I'd want the rest of the processing to continue

**Comment by liggitt** [comment]
comments addressed

File: pkg/cmd/util/clientcmd/clientcmd.go (hunk 2 above)
Comment (deads2k) [inline pkg/cmd/util/clientcmd/clientcmd.go]: Empty and unset are not the same thing.  Well, not in bash anyway, are they same in openshift?

Comment (liggitt) [inline pkg/cmd/util/clientcmd/clientcmd.go]: sadly, in env, they are the same

**Comment by deads2k** [comment]
[merge]

**Pull request merged**